great_expectations/execution_engine/execution_engine.py

```diff
@@ -527,6 +458,208 @@ def resolve_data_reference(
+                    MetricComputationConfiguration(
+                        metric_configuration=metric_to_resolve,
+                        metric_fn=metric_fn,
+                        metric_provider_kwargs=metric_provider_kwargs,
+                    )
+                )
+
+        return metric_fn_direct_configurations, metric_fn_bundle_configurations
+
+    def _get_computed_metric_evaluation_dependencies_by_metric_name(
+        self,
+        metric_to_resolve: MetricConfiguration,
+        metrics: Dict[Tuple[str, str, str], MetricValue],
+    ) -> Dict[str, Union[MetricValue, Tuple[Any, dict, dict]]]:
+        """
+        Gathers resolved (already computed) evaluation dependencies of metric-to-resolve (not yet computed)
+        "MetricConfiguration" object by "metric_name" property of resolved "MetricConfiguration" objects.
+
+        Args:
+            metric_to_resolve: dependent (not yet resolved) "MetricConfiguration" object
+            metrics: resolved (already computed) "MetricConfiguration" objects keyd by ID of that object
+
+        Returns:
+            Dictionary keyed by "metric_name" with values as computed metric or partial bundling information tuple
+        """
+        metric_dependencies_by_metric_name: Dict[
+            str, Union[MetricValue, Tuple[Any, dict, dict]]
+        ] = {}
+
+        metric_name: str
+        metric_configuration: MetricConfiguration
+        for (
+            metric_name,
+            metric_configuration,
+        ) in metric_to_resolve.metric_dependencies.items():
+            if metric_configuration.id in metrics:
+                metric_dependencies_by_metric_name[metric_name] = metrics[
+                    metric_configuration.id
+                ]
+            elif self._caching and metric_configuration.id in self._metric_cache:  # type: ignore[operator] # TODO: update NoOpDict
+                metric_dependencies_by_metric_name[metric_name] = self._metric_cache[
+                    metric_configuration.id
+                ]
+            else:
+                raise ge_exceptions.MetricError(
+                    message=f'Missing metric dependency: "{metric_name}" for metric "{metric_to_resolve.metric_name}".'
+                )
+
+        return metric_dependencies_by_metric_name
+
+    def _process_direct_and_bundled_metric_computation_configurations(
+        self,
+        metric_fn_direct_configurations: List[MetricComputationConfiguration],
+        metric_fn_bundle_configurations: List[MetricComputationConfiguration],
+    ) -> Dict[Tuple[str, str, str], MetricValue]:
+        """
+        This method processes directly-computable and bundled "MetricComputationConfiguration" objects.
+
+        Args:
+            metric_fn_direct_configurations: directly-computable "MetricComputationConfiguration" objects
+            metric_fn_bundle_configurations: bundled "MetricComputationConfiguration" objects (column aggregates)
```

Comment:
Let's pair offline, because I'm not following what the purpose of these two groups is.